ComputerInterface/CustomKeyboardKey.cs

```diff
@@ -74,61 +79,71 @@ public void Init(CustomComputer computer, EKeyboardKey key, Text keyboardText, s
         {
             Init(computer, key, keyboardText);
             if (keyboardText != null)
-			{
-				keyboardText.text = text;
-			}
+            {
+                keyboardText.text = text;
+            }
         }
 
         public void Init(CustomComputer computer, EKeyboardKey key, Text keyboardText, string text, Color buttonColor)
         {
             Init(computer, key, keyboardText, text);
             _material.color = buttonColor;
             _originalColor = buttonColor;
+
+            Color.RGBToHSV(buttonColor, out float H, out float S, out float _);
+            _pressedColor = Color.HSVToRGB(H, S, 0.6f);
         }
 
         private async void OnTriggerEnter(Collider collider)
         {
-            BumpIn();
-            if (_isOnCooldown) return;
-            _isOnCooldown = true;
-
-            if (collider.GetComponentInParent<GorillaTriggerColliderHandIndicator>() != null)
+            if (collider.TryGetComponent(out GorillaTriggerColliderHandIndicator component))
             {
-                GorillaTriggerColliderHandIndicator component = collider.GetComponent<GorillaTriggerColliderHandIndicator>();
+                if (_isOnCooldown) return;
+                _isOnCooldown = true;
 
+                BumpIn();
                 _computer.PressButton(this);
+                GorillaTagger.Instance.StartVibration(component.isLeftHand, GorillaTagger.Instance.tapHapticStrength / 2f, GorillaTagger.Instance.tapHapticDuration);
+                if (PhotonNetwork.InRoom && GorillaTagger.Instance.myVRRig != null)
+                    PhotonView.Get(GorillaTagger.Instance.myVRRig).RPC("PlayHandTap", RpcTarget.Others, 66, component.isLeftHand, 0.1f);
 
-                if (component != null)
-                {
-                    GorillaTagger.Instance.StartVibration(component.isLeftHand, GorillaTagger.Instance.tapHapticStrength / 2f, GorillaTagger.Instance.tapHapticDuration);
-                }
+                await Task.Delay(PRESS_COOLDOWN);
+                _isOnCooldown = false;
             }
-
-            await Task.Delay(PRESS_COOLDOWN);
-            _isOnCooldown = false;
         }
 
         private void OnTriggerExit(Collider collider)
         {
+            if (collider.GetComponent<GorillaTriggerColliderHandIndicator>() == null) return;
             BumpOut();
         }
 
         private void BumpIn()
         {
-            var pos = transform.localPosition;
-            pos.y -= KEY_BUMP_AMOUNT;
-            transform.localPosition = pos;
+            if (!_bumped)
+            {
+                _bumped = true;
+                var pos = transform.localPosition;
+                pos.y -= KEY_BUMP_AMOUNT;
+                transform.localPosition = pos;
+                collider.center -= new Vector3(0, 0, KEY_BUMP_AMOUNT / 1.125f);
```

Comment:
Why are you dividing by 1.125? I'm assuming its related to the local scale but I didn't notice any differences after removing it

Reply:
From adjusting a Unity project of the game, I felt like that change made for easier typing as that from the testing I did essentially made the collider stay in the same location whenever you pressed a key. This change was pre-beachmap update and I know buttons and probably keys were adjusted in some ways (Which includes Forest/Stump computer)